news-TS/src/components/controller/loader.ts

```diff
@@ -0,0 +1,51 @@
+import { apiRequest, apiOptions, endpoints, drawContentFunc, httpErr, IApiResponse } from '../../types/index';
+
+class Loader {
```

Comment:
I don't know if it your code or you just moved it to ts but I don't see much value in this file. It has a lot of unecessery and complicated code, instead I would use some sort of api client modules which are just set of functions that calls some apis. For example: getSources, getTopHeadlines, getEverything and use them directly.